### PR TITLE
Vault: upload artifacts to s3

### LINKED ISSUE #1351: Vault: upload artifacts to s3
## Task Description
Vault runs without the user who is running the vault do not have access to files created on the vault server.

In a 'vault only run' with a computation that outputs file artifacts (vbm), those files are only available on the vault server. After a pipeline finishes these should be available (via the api) to anyone who has permissions to them.

## Task Notes
* change headless to vault in packaing and docs :), also change `cloud` to`vault` for naming sanity
* run a vault client run with VBM (stats) this will give you file output on finish
* on vault client finish (on a run with only vault clients??) upload everything in the `output/` dir to s3, using runid as a lookup key
* delete these artifacts locally on successful upload
* make an api call to store these in the db, perhaps in the run table?
* proxy the s3 download through hapi, there are some examples of this online (they might be express but it should be roughly the same), this puts access control to the s3 buckets in the hands of the api
* make a front end for this, likely somewhere w the `view results` button on the results page and widgets

### LINKED ISSUE #1366: UI: download and display s3 vault data
## Task Description
When a vault run has no normal users, the vault will upload the results to s3. The UI then needs to grab and unpack this for the user

## Task Notes
* download the resource based on the `runid` - this may not be available yet while it upload
* the `open local results` button should be `download results` if its a vault only run
* unpack the results as if there were normal results eg `COINSTAC_HOME/consortiumID/runID/{results}`
* the results page may need some work to not crash if the results are not available (before download)

### PR DESCRIPTION
**Pull request by dylanmartin**
* **Please check if the PR fulfills these requirements**
- [x] Passes e2e testing
- [x] Docs have been added / updated if necessary (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Feature

* **What is the current behavior?** (You can also link to an open issue here)
closes _issue #1358 #1351 #1350

* **What is the new behavior (if this is a feature change)?**
When a pipeline runs with vault users only, one of the vault users will upload the run assets to s3 upon completion.

* **Other information**:
A next step will be to allow human users to download and view results in the COINSTAC ui

### CODE REVIEW

**Comment by dylanmartin** [comment]
@rssk we still need to talk about changing "headless" to "vault" throughout the codebase.

**Comment by dylanmartin** [comment]
https://github.com/trendscenter/coinstac/issues/1351#issuecomment-1165878983

File: packages/coinstac-api-server/src/routes/cloudFiles.js [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+          const result = await downloadFromS3(fileKey);
+          return h.response(result.Body).code(200);
+        } catch (e) {
+          console.log(e);
+          return h.response(e).code(500);
+        }
+
+      
+      },
+      payload: {
```
Comment (dylanmartin) [inline packages/coinstac-api-server/src/routes/cloudFiles.js:93]: Try to upload and download a big file. 500mb

**Comment by rssk** [comment]
Can you check if there are files to upload in the output directory?

File: packages/coinstac-client-core/src/index.js [hunk 2]
```diff
@@ -435,6 +476,42 @@ class CoinstacClient {
...
+    axios.post(
+      `${process.env.API_URL}/uploadFiles`,
+      formData,
+      postConfig
+    ).then((result) => {
+      console.log(result);
+    }).catch((e) => {
+      console.log(e);
+    });
+
```
Comment (rssk) [inline packages/coinstac-client-core/src/index.js]: core has a logger instance that should be used for logs

File: packages/coinstac-ui/app/main/index.js [hunk 3]
```diff
@@ -694,5 +699,100 @@ loadConfig()
...
+        });
+
+        // extract the tar
+        const readStream = fs.createReadStream(outputFilePath);
+        const writeStream = tar.extract(runOutputDirectory);
+        readStream.pipe(gunzip()).pipe(writeStream);
+
+        await new Promise((resolve, reject) => {
+          writeStream.on('finish', () => {
+            console.log('writeStream finished');
```
Comment (rssk) [inline packages/coinstac-ui/app/main/index.js]: use the logger here too;)

File: packages/coinstac-ui/app/render/components/results/result.jsx [hunk 4]
```diff
@@ -213,7 +224,36 @@ class Result extends Component {
...
+                  try {
+                    const result = await ipcRenderer.invoke('download-run-assets', {
+                      runId: run.id, authToken, clientId, apiServerUrl,
+                    });
+                    console.log(result);
+                    this.setState({ downloading: false });
+                    await this.doFilesExist(run.id);
+                  } catch (e) {
+                    console.log(e);
+                    this.setState({ downloading: false });
```
Comment (rssk) [inline packages/coinstac-ui/app/render/components/results/result.jsx]: I'm guessing the top log is for debug, but the bottom should probably be wired to an error toast/notification

**Review by rssk — CHANGES_REQUESTED** [review]

**Review by rssk — APPROVED** [review]

**Pull request merged**